src/libpriv/rpmostree-core.c

```diff
@@ -1011,31 +1311,51 @@ rpmostree_context_prepare_install (RpmOstreeContext    *self,
  *
  * This can be used to efficiently see if the goal has changed from a
  * previous one.
+ *
+ * It can also handle goals in which only dummy header-only pkgs are
+ * available. In such cases, it uses the ostree checksum of the pkg
+ * instead.
  */
 void
-rpmostree_dnf_add_checksum_goal (GChecksum *checksum,
-                                 HyGoal     goal)
+rpmostree_dnf_add_checksum_goal (GChecksum   *checksum,
+                                 HyGoal       goal,
+                                 GHashTable  *pkg_to_checksum)
 {
   g_autoptr(GPtrArray) pkglist = NULL;
-  guint i;
-  g_autoptr(GPtrArray) pkg_checksums = g_ptr_array_new_with_free_func (g_free);
+  g_autoptr(GPtrArray) pkg_checksums =
+    g_ptr_array_new_with_free_func (g_free);
 
   pkglist = hy_goal_list_installs (goal, NULL);
   g_assert (pkglist);
-  for (i = 0; i < pkglist->len; i++)
+  for (guint i = 0; i < pkglist->len; i++)
     {
       DnfPackage *pkg = pkglist->pdata[i];
-      int pkg_checksum_type;
-      const unsigned char *pkg_checksum_bytes = dnf_package_get_chksum (pkg, &pkg_checksum_type);
-      g_autofree char *pkg_checksum = hy_chksum_str (pkg_checksum_bytes, pkg_checksum_type);
-      char *pkg_checksum_with_type = g_strconcat (hy_chksum_name (pkg_checksum_type), ":", pkg_checksum, NULL);
-      
-      g_ptr_array_add (pkg_checksums, pkg_checksum_with_type);
+
+      if (pkg_to_checksum && g_hash_table_contains (pkg_to_checksum, pkg))
+        {
+          /* the ostree checksum itself is already a good representation
+           * of the whole package */
```

Comment:
Hmm...but it includes a timestamp.  We should probably make the commit timestamp derive from some timestamp in the rpm header?